python/ray/train/_checkpoint.py

```diff
@@ -129,7 +130,7 @@ def get_metadata(self) -> Dict[str, Any]:
 
         If no metadata is stored, an empty dict is returned.
         """
-        metadata_path = os.path.join(self.path, _METADATA_FILE_NAME)
+        metadata_path = (Path(self.path) / _METADATA_FILE_NAME).as_posix()
```

Comment:
Nit: I used to like the `/`, but I now think all the parentheses makes it annoying, so I prefer `Path(..., ...)` or `Path(...).joinpath(...).as_posix()` 😄 

```suggestion
        metadata_path = Path(self.path,  _METADATA_FILE_NAME).as_posix()
```